packages/utils/tool-utils/src/snapshotNormalizer.ts

```diff
@@ -110,39 +111,54 @@ function getNormalizedBlobContent(blobContent: string, blobName: string): string
 export function getNormalizedSnapshot(snapshot: ITree, config?: ISnapshotNormalizerConfig): ITree {
     // Merge blobs to normalize in the config with runtime blobs to normalize. The contents of these blobs will be
     // parsed and deep sorted.
-    const blobsToNormalize = [ ...config?.blobsToNormalize ?? [] ];
     const normalizedEntries: ITreeEntry[] = [];
 
     for (const entry of snapshot.entries) {
-        switch (entry.type) {
-            case TreeEntry.Blob: {
-                let contents = entry.value.contents;
-                // If this blob has to be normalized or it's a GC blob, parse and sort the blob contents first.
-                if (blobsToNormalize.includes(entry.path) || entry.path.startsWith(gcBlobPrefix)) {
-                    contents = getNormalizedBlobContent(contents, entry.path);
-                }
-                normalizedEntries.push(new BlobTreeEntry(entry.path, contents));
-                break;
-            }
-            case TreeEntry.Tree: {
-                normalizedEntries.push(new TreeTreeEntry(entry.path, getNormalizedSnapshot(entry.value, config)));
-                break;
-            }
-            case TreeEntry.Attachment: {
-                normalizedEntries.push(new AttachmentTreeEntry(entry.path, (entry.value).id));
-                break;
-            }
-
-            default:
-                throw new Error("Unknown entry type");
-        }
+        normalizedEntries.push(normalizeEntry(entry, config));
     }
 
-    // Sory the tree entries based on their path.
+    // Sort the tree entries based on their path.
     normalizedEntries.sort((a, b) => a.path.localeCompare(b.path));
 
     return {
         entries: normalizedEntries,
         id: snapshot.id,
     };
 }
+
+function normalizeEntry(
+    entry: ITreeEntry,
+    config: ISnapshotNormalizerConfig | undefined,
+): ITreeEntry {
+    switch (entry.type) {
+        case TreeEntry.Blob: {
+            let contents = entry.value.contents;
+            // If this blob has to be normalized or it's a GC blob, parse and sort the blob contents first.
+            if (config?.blobsToNormalize?.includes(entry.path) || entry.path.startsWith(gcBlobPrefix)) {
+                contents = getNormalizedBlobContent(contents, entry.path);
+            }
+            return new BlobTreeEntry(entry.path, contents);
+        }
+        case TreeEntry.Tree: {
+            if(config?.excludedChannelContentTypes !== undefined) {
```

Comment:
this is the core handling for the non comparable channels. if we fine one, we exclude all the blobs except for the attributes.

Reply:
the rest of the function is the same as above

Reply:
Shouldn't the `config.excludedChannelContentTypes.includes(type)` check happen before we start looking at its content. That way we won't do it at the data store level but only at the DDS level.

Reply:
this is just an optimization were we skip looking for the attributes blob if there are no excludedChannelContentTypes

Reply:
Oh I get it now. The values in `excludedChannelContentTypes` are the channel types and not channel ids. I thought it was ids and if so we could have compared the id before we looked at the attributes. But the type info is in the ".attributes" blob so we have to look at it.